test/gtest/ucp/test_ucp_am.cc

```diff
@@ -692,6 +700,69 @@ class test_ucp_am_nbx : public test_ucp_am_base {
     ucp_mem_h                       m_rx_memh;
 };
 
+class test_ucp_am_id : public test_ucp_am_nbx {
+protected:
+    void reset_counters()
+    {
+        test_ucp_am_nbx::reset_counters();
+        m_recv_counter_cb_1 = 0;
+        m_recv_counter_cb_2 = 0;
+    }
+
+    void test_am_id_handler()
+    {
+        reset_counters();
+
+        EXPECT_UCS_OK(set_am_data_handler_internal(
+                                        receiver(), 0x1,
+                                        am_id_overflow_data_cb_1, this));
+        EXPECT_EQ(UCS_ERR_INVALID_PARAM, set_am_data_handler_internal(
+                                         receiver(), 0xffff0001,
+                                         am_id_overflow_data_cb_2, this));
+
+        ucp_request_param_t param;
+        param.op_attr_mask = 0;
+
+        ucs_status_ptr_t sptr = update_counter_and_send_am(
+                                              NULL, 0ul, NULL, 0, 0x1, &param);
+        wait_receives();
+        ASSERT_UCS_OK(request_wait(sptr));
+        EXPECT_EQ(m_recv_counter_cb_1, 1);
+        EXPECT_EQ(m_recv_counter_cb_2, 0);
+
+        sptr = update_counter_and_send_am(NULL, 0ul, NULL, 0, 0xffff0001, &param);
+        EXPECT_EQ(UCS_PTR_STATUS(sptr), UCS_ERR_INVALID_PARAM);
+        EXPECT_EQ(m_recv_counter_cb_1, 1);
+        EXPECT_EQ(m_recv_counter_cb_2, 0);
```

Comment:
I don't think we need to test the "good" flow, there are quite a lot of ucp ap tests already, in the new tests we can just check the erroneous cases of passing invalid am id